File: programming/problem-solving/python/problem-solving/src/no_1_two_sum/twoSum.py
def twoSum(nums, target):
    dictStore = {}
    for index, item in enumerate(nums):
        now = item
        looking_for = target - now
        found = dictStore.get(looking_for)
        if found != None and found > -1:
            return [dictStore.get(looking_for), index]
        else:
            dictStore[now] = index
    return [-1, -1]


print(twoSum([3, 1, 2, 4], 3))

File: programming/problem-solving/javascript/problem-solving/src/no_1_two_sum/two_sum.js
const twoSum = (nums, target) => {
    const object = {};
    for (var i = 0; i < nums.length; i++) {
        const now = nums[i];
        const lookingFor = target - now;
        if (object[lookingFor] >= -1) {
            return [object[lookingFor], i];
        } else
            object[now] = i;

    }
    return [-1, -1];
}

console.log(twoSum([4, 1, 2, 3], 3));